ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -302,24 +307,28 @@ runDataDiffusion tracers
                Left sd -> do
                    a <- Socket.getSocketName sd
                    case a of
-                        (Socket.SockAddrUnix path) ->
+                        (Socket.SockAddrUnix path) -> do
+                          traceWith dtSocketTracer $ SocketTrace "Creating local server socket for unix path" (Just path)
                           return (sd, Snocket.localSnocket iocp path)
                         _                          ->
                             -- TODO: This should be logged.
                             throwIO UnsupportedLocalSocketType
 #endif
-               Right a -> do
-                   let sn = Snocket.localSnocket iocp a
-                   sd <- Snocket.open sn (Snocket.addrFamily sn $ Snocket.localAddressFromPath a)
+               Right path -> do
+                   let sn = Snocket.localSnocket iocp path
+                   traceWith dtSocketTracer $ SocketTrace "Creating local server socket for snocket path" (Just path)
+                   sd <- Snocket.open sn (Snocket.addrFamily sn $ Snocket.localAddressFromPath path)
+                   traceWith dtSocketTracer $ SocketTrace "Created local server socket for snocket path" (Just path)
                    return (sd, sn)
         )
         (\(sd,sn) -> Snocket.close sn sd) -- We close the socket here, even if it was provided for us.
         (\(sd,sn) -> do
 
           case daLocalAddress of
                Left _ -> pure () -- If a socket was provided it should be ready to accept
-               Right a -> do
-                 Snocket.bind sn sd $ Snocket.localAddressFromPath a
+               Right path -> do
+                 traceWith dtSocketTracer $ SocketTrace "Binding to socket for path: " (Just path)
```

Comment:
`Binding socket to path: ...` would sound better.

Reply:
Note: you can also log the socket file descriptor with `show sd` or if you just need the file descriptor you can use `socketFd sd :: IO CInt`.

Reply:
I wasn't able to find `socketFd` so I went with `show`.

Reply:
Sorry, it's `socketToFd`

Reply:
Looks like `socketToFd` is only available on `network-3.1.2.0`, but we are stuck on `network-3.1.0.1` for the moment.